src/util.ts

```diff
@@ -17,25 +18,42 @@ export function log(message: string, ...optionalContext: any[]) {
 }
 
 /**
- * @param array1
- * @param array2
+ * @param seed
  */
-export function concatUint8Arrays(array1: Uint8Array, array2: Uint8Array): Uint8Array {
-  const result = new Uint8Array(array1.length + array2.length);
-  result.set(array1);
-  result.set(array2, array1.length);
-  return result;
+export function genKeyPairFromSeed(seed: Uint8Array): KeyPair {
+  const bytes = new Uint8Array([...sha512(stringToUint8ArrayUtf8("root discoverable key")), ...sha512(seed)]);
```

Comment:
explicit `stringToUint8ArrayUtf8` not needed here, `sha512` already does that internally
```suggestion
  const bytes = new Uint8Array([...sha512("root discoverable key"), ...sha512(seed)]);
```